waitforx/waitforx.c

```diff
@@ -0,0 +1,96 @@
+#include <X11/extensions/Xrandr.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/signal.h>
+#include <unistd.h>
+
+#define ATTEMPTS 10
+#define ALARM_WAIT 30
+
+void
+alarm_handler(int signal_num)
+{
+    printf("Unable to find RandR outputs after %d seconds\n", ALARM_WAIT);
```

Comment:
Can't use `printf()` in a signal handler. See [signal-safety(7)](https://man7.org/linux/man-pages/man7/signal-safety.7.html)

Given this is unlikely to be used on Windows, here's a suggested alternative.

```c
/* Avoid printf() in signal handler (see signal-safety(7)) */
const char msg[] = "Timed out waiting for RandR outputs\n";
g_file_write(1, msg, g_strlen(msg));
```